src/app/AttributePathParams.h

```diff
@@ -65,10 +63,30 @@ struct AttributePathParams
     inline bool HasWildcardAttributeId() const { return mAttributeId == kInvalidAttributeId; }
     inline bool HasWildcardListIndex() const { return mListIndex == kInvalidListIndex; }
 
-    EndpointId mEndpointId   = kInvalidEndpointId;
-    ClusterId mClusterId     = kInvalidClusterId;
-    AttributeId mAttributeId = kInvalidAttributeId;
-    ListIndex mListIndex     = kInvalidListIndex;
+    bool IsAttributePathSupersetOf(const AttributePathParams & other) const
+    {
+        VerifyOrReturnError(HasWildcardEndpointId() || mEndpointId == other.mEndpointId, false);
+        VerifyOrReturnError(HasWildcardClusterId() || mClusterId == other.mClusterId, false);
+        VerifyOrReturnError(HasWildcardAttributeId() || mAttributeId == other.mAttributeId, false);
+        VerifyOrReturnError(HasWildcardListIndex() || mListIndex == other.mListIndex, false);
+
+        return true;
+    }
+
+    bool IsAttributePathSupersetOf(const ConcreteAttributePath & other) const
+    {
+        VerifyOrReturnError(HasWildcardEndpointId() || mEndpointId == other.mEndpointId, false);
+        VerifyOrReturnError(HasWildcardClusterId() || mClusterId == other.mClusterId, false);
+        VerifyOrReturnError(HasWildcardAttributeId() || mAttributeId == other.mAttributeId, false);
+
+        return true;
```

Comment:
Even if we have a non-wildcard list index?  That seems odd.

Reply:
so we also need list index? Per jerry, it seems we don't support list index, so i am planning to remove all list index.

Reply:
Works for me.